k8s-dev/cstor-pool-mgmt/deployment-cstor-pool.yaml

```diff
@@ -0,0 +1,72 @@
+# The deployment for cstor-pool and cstor-pool-mgmt.
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: cstor-pool
+  labels:
+    app: cstor-pool
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: cstor-pool
+  template:
+    metadata:
+      labels:
+        app: cstor-pool
+    spec:
+      containers:
+      - name: cstor-pool
+#The image name here will be changed to openebs once the business logic pr is reviewed and merged in openebs/maya.
+        image: gkganeshr/cstor-pool:v1.1
+        ports:
+        - containerPort: 80
+        securityContext: 
+          privileged: true
+
+        volumeMounts:
+        - name: device
+          mountPath: /dev     
+        - name: tmp
+          mountPath: /tmp
+#To avoid clash between terminating and restarting pod, we keep initial delay.
+        lifecycle:
+          postStart:
+            exec:
+              command: ["/bin/sh", "-c", "sleep 15"]
+
+      - name:    cstor-pool-mgmt
+#The image name here will be changed to openebs once the business logic pr is reviewed and merged in openebs/maya.
+        image: gkganeshr/cstor-pool-mgmt:v1.8
+        ports:
+        - containerPort: 80
+        securityContext: 
+          privileged: true
```

Comment:
do we need privileged mode for sidecar container?

Reply:
we are trying to access disks, /dev and it needs privileged mode to avoid permission errors.

Reply:
you are right. I forgot that sidecar touches the device files as well.